static/app/views/issueList/issueViewsHeader.tsx

```diff
@@ -192,6 +197,54 @@ function IssueViewsIssueListHeaderTabsContent({
     queryParams,
   ]);
 
+  const {mutate: updateViews} = useUpdateGroupSearchViews();
+
+  const debounceUpdateViews = useMemo(
+    () =>
+      debounce((groupSearchViews: UpdateGroupSearchViewPayload[]) => {
```

Comment:
Oh I also realized an issue with debouncing the request. If you change your page filters, then modify your query and save before the debounce time is over, this request will override the changes you made in the interim. Maybe we'll just have to deal with the excess requests for now